packages/schema/src/config/common.ts

```diff
@@ -175,7 +175,31 @@ export default defineUntypedSchema({
    * ```
    */
   buildDir: {
-    $resolve: async (val: string | undefined, get): Promise<string> => resolve(await get('rootDir') as string, val || '.nuxt'),
+    $resolve: async (val: string | undefined, get): Promise<string> => {
+      const rootDir = await get('rootDir') as string
+
+      if (val) {
+        return resolve(rootDir, val)
+      }
+
+      const defaultBuildDir = resolve(rootDir, '.nuxt')
+
+      const isDev = await get('dev') as boolean
+      if (isDev) {
+        return defaultBuildDir
+      }
+
+      const buildId = await get('buildId') as string
+
+      // TODO: nuxi CLI should ensure .nuxt dir exists
+      if (!existsSync(defaultBuildDir)) {
+        // This is to ensure that types continue to work for CI builds
+        return defaultBuildDir
+      }
+
+      // TODO: handle build caching
+      return resolve(rootDir, 'node_modules/.cache/nuxt/builds', buildId)
```

Comment:
(maybe in later pr) we probably need to take care about GC. Size of this dir can be so big over the time. One fast start might be to keep `.cache/nuxt/builds.json` with id + date (so we can gc)

Reply:
Agreed!